torch/csrc/autograd/engine.cpp

```diff
@@ -1032,7 +974,6 @@ bool Engine::is_checkpoint_valid() {
 void Engine::init_local_ready_queue(std::shared_ptr<ReadyQueue> ready_queue) {
   if (ready_queue) {
     // if ready_queue provided in the caller, use the caller's ready_queue to initialize local_ready_queue
-    TORCH_INTERNAL_ASSERT(!local_ready_queue);
```

Comment:
Why do you need to remove that? This should always be true, even from the distributed call no? Otherwise you're doing reentrant for a dist engine? :D

Reply:
In the distributed engine the `execute_graph_task_until_rq_empty` is called from a thread pool (RPC threadpool or JIT threadpool), so if the thread pool launches the function twice or more, the local_ready_queue already been set to the ready queue that's created by previous calls. That's why I remove the assert here.

Reply:
Ok,
I might be cleaning this all logic later. Will be fine for now.

Reply:
actually I did a couple of runs with initializing the local_ready_queue to support reentrant backwards, it turns out the perf becomes bad after that (around 10% drop), this is pretty surprising as I wouldn't expect a big drop like this. So I reverted this change and added a TODO in the comment to investigate further in the future.